core/src/main/java/org/apache/accumulo/core/fate/FateExecutor.java

```diff
@@ -105,15 +105,22 @@ public FateExecutor(Fate<T> fate, T environment, Set<Fate.FateOperation> fateOps
   protected void resizeFateExecutor(Map<Set<Fate.FateOperation>,Integer> poolConfigs,
       long idleCheckIntervalMillis) {
     final var pool = transactionExecutor;
-    final var runningTxRunners = getRunningTxRunners();
+    final var runningTxRunnersCopy = getRunningTxRunners();
     final int configured = poolConfigs.get(fateOps);
     ThreadPools.resizePool(pool, () -> configured, poolName);
-    final int needed = configured - runningTxRunners.size();
+    final int needed = configured - runningTxRunnersCopy.size();
     if (needed > 0) {
       // If the pool grew, then ensure that there is a TransactionRunner for each thread
       for (int i = 0; i < needed; i++) {
         try {
-          pool.execute(new TransactionRunner());
+          final TransactionRunner tr = new TransactionRunner();
+          synchronized (runningTxRunners) {
+            if (pool.isShutdown() || pool.isTerminating()) {
+              return;
+            }
+            runningTxRunners.add(tr);
+            pool.execute(tr);
+          }
```

Comment:
I like this switch to add the runner before calling execute, this is safer.

Some comments:
1) Should we maybe just avoid the use of the copy and just lock the runningTxRunners for the entire if-elseif-else block? Using the copy and the real seems a bit sketchy, but I'm not sure if it's actually wrong/a problem. The copy was sketching me out when I was working on #5813 so I removed it there as well.
2) `pool.isShutdown() || pool.isTerminating()` --> `pool.isShutdown()` might be clearer. I'm pretty sure `pool.isTerminating()` is true only if `pool.isShutdown()` is true

Reply:
Seems like it may be good to get 5813 merged and then make this change to avoid the copy change.

Checking the pool for shutdown here has a race condition like in the following.

```java

            if (pool.isShutdown() || pool.isTerminating()) {
              return;
            }
            runningTxRunners.add(tr);
            // pool could be shutdown here and a rejected excecution exception thrown
            pool.execute(tr);
```

So maybe could do something like the following.

```java
try {
   runningTxRunners.add(tr);
   pool.execute(tr);
}catch (RejectedExecutionException e){
   runningTxRunners.remove(tr);
}
```